packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1371,11 +1407,25 @@ export class KeyringController extends BaseController<
   /**
    * Verifies the that the seed phrase restores the current keychain's accounts.
    *
+   * @param keyringId - The id of the keyring to verify.
    * @returns Promise resolving to the seed phrase as Uint8Array.
    */
-  async verifySeedPhrase(): Promise<Uint8Array> {
+  async verifySeedPhrase(keyringId?: string): Promise<Uint8Array> {
     this.#assertIsUnlocked();
-    return this.#withControllerLock(async () => this.#verifySeedPhrase());
+    const keyring = this.#getKeyringByIdOrDefault(keyringId);
```

Comment:
similarly to the `exportSeedPhrase` method comment: I think that defaulting to the primary keyring could be unexpected or even dangerous, because we would verify a potentially completely different keyring than what the consumer expects